lib/chef/provider/template/content.rb

```diff
@@ -36,7 +36,26 @@ def template_location
         private
 
         def file_for_provider
-          context = TemplateContext.new(new_resource.variables)
+          # Deal with any DelayedEvaluator values in the template variables.
+          visitor = lambda do |obj|
+            case obj
+            when Hash
+              # Avoid mutating hashes in the resource in case we're changing anything.
+              obj.each_with_object(obj.class.new) do |(key, value), memo|
+                memo[key] = visitor.call(value)
+              end
+            when Array
+              # Avoid mutating arrays in the resource in case we're changing anything.
+              obj.map { |value| visitor.call(value) }
+            when DelayedEvaluator
+              new_resource.instance_eval(&obj)
+            else
+              obj
+            end
+          end
```

Comment:
this smells like you could make it into a chef/mixin/visitor, didn't you just do something very similar for the ohai freezing, and i think i just asked you to try to do something similar for properties freezing?